test/integration/setup.go

```diff
@@ -189,8 +200,65 @@ func updateProtoGolden(resp protoreflect.ProtoMessage, fname string) {
 		log.Printf("could not write golden files to %s", fname)
 		return
 	}
-	err = ioutil.WriteFile(fname, jsonByte, 0644)
+	if len(shard) == 1 && shard[0] {
+		err = writeJsonShard(jsonByte, fname)
+	} else {
+		err = ioutil.WriteFile(fname, jsonByte, 0644)
+	}
 	if err != nil {
 		log.Printf("could not write golden files to %s", fname)
 	}
 }
+
+func writeJsonShard(jsonByte []byte, fname string) error {
+	jsonLines := strings.Split(string(jsonByte), "\n")
+	for i := 0; ; i++ {
+		start := i * lineLimit
+		end := (i + 1) * lineLimit
+		if end > len(jsonLines) {
+			end = len(jsonLines)
+		}
+		err := ioutil.WriteFile(
+			fmt.Sprintf("%s.%03d", fname, i),
+			[]byte(strings.Join(jsonLines[start:end], "\n")),
+			0644,
+		)
+		if err != nil {
+			return err
+		}
+		if end == len(jsonLines) {
+			break
+		}
+	}
+	return nil
+}
+
+func find(path, fname string) ([]string, error) {
+	var result []string
+	err := filepath.WalkDir(path, func(s string, d fs.DirEntry, e error) error {
+		if e != nil {
+			return e
+		}
+		if strings.Contains(s, fname) {
+			result = append(result, s)
+		}
+		return nil
+	})
+	return result, err
```

Comment:
Is result guaranteed to be sorted on name?  Document assumption?

Reply:
Yea, the output is sorted, added comment